### PR TITLE
Add Not Operator Functionality

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [X] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [ ] 1
- [X] 2
- [ ] 3

**What this does:**
Add functionality for `not` operator for logical conditions.

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

File: nucypher/policy/conditions/lingo.py [hunk 1]
```diff
@@ -111,9 +129,25 @@ def __init__(
...
+
+        if operator == self.NOT_OPERATOR:
+            if len(operands) != 1:
+                raise InvalidCondition(
+                    f"Only 1 operand permitted for '{operator}' condition"
+                )
+        elif len(operands) < 2:
+            raise InvalidCondition(
+                f"Minimum of 2 operand needed for '{operator}' compound condition"
+            )
```
Comment (cygnusv) [inline nucypher/policy/conditions/lingo.py:148]: This block is similar to `validate_operator_and_operands()` above. Can this be reused?

File: nucypher/policy/conditions/lingo.py [hunk 2]
```diff
@@ -134,12 +168,17 @@ def verify(self, *args, **kwargs) -> Tuple[bool, Any]:
                 # short-circuit check
                 if overall_result is False:
                     break
-            else:
+            elif self.operator == self.OR_OPERATOR:
                 # or operator
```
Comment (cygnusv) [inline nucypher/policy/conditions/lingo.py:172]:
```suggestion
```

**Review by cygnusv — APPROVED** [review]

File: nucypher/policy/conditions/lingo.py (hunk 1 above)
Comment (derekpierre) [reply nucypher/policy/conditions/lingo.py]: ✔️

File: nucypher/policy/conditions/lingo.py (hunk 2 above)
Comment (derekpierre) [reply nucypher/policy/conditions/lingo.py]: ✔️

**Review by piotr-roslaniec — APPROVED** [review]

File: tests/integration/learning/test_domains.py [hunk 3]
```diff
@@ -42,6 +42,7 @@ def registry_2(domain_2, test_registry):
     return ContractRegistry(MockRegistrySource(domain=domain_2))
 
 
+@pytest.mark.skip("inconsistent behaviour on CI - see #3289")
```
Comment (KPrasch) [inline tests/integration/learning/test_domains.py:45]: This is an important regression test, I do not think it's good to keep it skipped for long.  #3289

**Review by KPrasch — APPROVED** [review]

**Pull request merged**